src/main/java/seedu/address/model/student/Year.java

```diff
@@ -25,17 +26,26 @@
 
     /**
      * Constructs a {@code Year}.
-     *
-     * @param schoolType A valid school type
-     * @param level A valid level
      */
-    public Year(SchoolType schoolType, Integer level) {
-        requireNonNull(schoolType);
-        requireNonNull(level);
+    public Year(String year) {
+        requireNonNull(year);
+        final Matcher matcher = YEAR_FORMAT.matcher(year.trim());
+        assert matcher.matches();
+        String schoolTypeString = matcher.group("school").trim().toLowerCase();
+        String levelString = matcher.group("level").trim().toLowerCase();
+
+        SchoolType schoolType = getSchoolType(schoolTypeString);
+        Integer level = Integer.parseInt(levelString);
         this.schoolType = schoolType;
         this.level = level;
     }
 
+    private static SchoolType getSchoolType(String schoolTypeString) {
+        String trimmedSchool = schoolTypeString.trim();
+        assert SchoolType.isValidSchoolType(trimmedSchool);
```

Comment:
And also I think note that `isValidSchoolType` is actually less stringent than the validation regex in `Year.java` due to casing (so 'sEC 3' would be valid under `isValidSchoolType` but not under `Year.java`). It doesn't affect anything here but if more changes are made this might potentially be relevant?

Reply:
I wouldn't worry about this as this assertion only happens after the input string matches `Year.VALIDATION_REGEX`. Hence if it passes the earlier assertion in the constructor, it will definitely pass this one. This is just another safety net.